api/handler/handler.go

```diff
@@ -337,3 +341,30 @@ func (h *ShortenerHandler) GenerateShorterLinkPOSTBatch(w http.ResponseWriter, r
 		return
 	}
 }
+
+func (h *ShortenerHandler) BatchDeleteLinks(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	bodyStr := string(body)
+	bodyStr = strings.ReplaceAll(bodyStr, "[", "")
+	bodyStr = strings.ReplaceAll(bodyStr, "]", "")
+	bodyStr = strings.ReplaceAll(bodyStr, "\"", "")
+	linksToDelete := strings.Split(bodyStr, ",")
+	for _, link := range linksToDelete {
+		log.Println(link)
+	}
+	userID := ctx.Value("UserID").(uint64)
+	userIDStr := strconv.FormatUint(userID, 10)
+	go func() {
+		h.Shortener.Storage.SetDeleteFlag(linksToDelete, userIDStr)
```

Comment:
Можно было бы обернуть вызов Storage.SetDeleteFlag(linksToDelete, userIDStr) в отдельный метод сервиса, чтобы не вытаскивать наружу Storage, на случай если метод будет переиспользоваться
```suggestion
		h.Shortener.SetDeleteFlag(linksToDelete, userIDStr)
```